snap/snapcraft.yaml

```diff
@@ -20,82 +18,109 @@ architectures:
+      SECRETSTORE_TOKENFILE: $SNAP_DATA/device-rfid-llrp/secrets-token.json
     daemon: simple
+    passthrough:
+      install-mode: disable
     plugs: [network, network-bind]
 
+plugs:
+  device-config:
+    interface: content 
+    content: device-config
+    target: $SNAP_DATA/config/device-rfid-llrp/res
+
 parts:
-  go:
+  version:
     plugin: nil
     source: snap/local
-    build-packages: [curl]
+    override-pull: |
+      cd $SNAPCRAFT_PROJECT_DIR
+      GIT_VERSION=$(git describe --tags --abbrev=0 | sed 's/v//')
+      if [ -z "$GIT_VERSION" ]; then
+        GIT_VERSION="0.0.0"
+      fi
+      snapcraftctl set-version ${GIT_VERSION}
+      
+  hooks:
+    source: snap/local/hooks
+    plugin: make
+    build-snaps:
+      - go/1.16/stable
     override-build: |
-      # use dpkg architecture to figure out our target arch
-      # note - we specifically don't use arch
-      case "$(dpkg --print-architecture)" in
-        amd64)
-          FILE_NAME=go1.15.2.linux-amd64.tar.gz
-          FILE_HASH=b49fda1ca29a1946d6bb2a5a6982cf07ccd2aba849289508ee0f9918f6bb4552
-          ;;
-        arm64)
-          FILE_NAME=go1.15.2.linux-arm64.tar.gz
-          FILE_HASH=c8ec460cc82d61604b048f9439c06bd591722efce5cd48f49e19b5f6226bd36d
-          ;;
-      #  armhf)
-      #    FILE_NAME=go1.15.2.linux-armv6l.tar.gz
-      #    FILE_HASH=c12e2afdcb21e530d332d4994919f856dd2a676e9d67034c7d6fefcb241412d9
-      #    ;;
-      #  i386)
-      #    FILE_NAME=go1.15.2.linux-386.tar.gz
-      #    FILE_HASH=5a91080469df6b91f1022bdfb0ca75e01ca50387950b13518def3d0a7f6af9f1
-      #    ;;
-      esac
-      # download the archive, failing on ssl cert problems
-      curl https://dl.google.com/go/$FILE_NAME -O
-      echo "$FILE_HASH $FILE_NAME" > sha256
-      sha256sum -c sha256 | grep OK
-      tar -C $SNAPCRAFT_STAGE -xf go*.tar.gz --strip-components=1
-    prime:
-      - "-*"
-
-  device-rfid-llrp-go:
+      cd $SNAPCRAFT_PART_SRC
+      make build
+      install -DT ./cmd/configure/configure "$SNAPCRAFT_PART_INSTALL/snap/hooks/configure"
+      install -DT ./cmd/install/install "$SNAPCRAFT_PART_INSTALL/snap/hooks/install"
+
+  device-rfid-llrp:
     source: .
     plugin: make
-    build-packages: [git]
-    after: [go]
+    build-packages: [git, libzmq3-dev, pkg-config]
+    stage-packages: [libzmq5]
+    build-snaps:
+      - go/1.16/stable
     override-build: |
       cd $SNAPCRAFT_PART_SRC
+
+      # The makefile uses a VERSION file to set the Version string. 
+      # This file needs to be generated here
+      GIT_VERSION=$(git describe --tags --abbrev=0 | sed 's/v//')
+      if [ -z "$GIT_VERSION" ]; then
+        GIT_VERSION="0.0.0"
+      fi
+      echo $GIT_VERSION > ./VERSION
+
+      go mod tidy
       make build
 
-      install -DT "./cmd/device-rfid-llrp-go" "$SNAPCRAFT_PART_INSTALL/bin/device-rfid-llrp-go" 
+      install -DT "./cmd/device-rfid-llrp" "$SNAPCRAFT_PART_INSTALL/bin/device-rfid-llrp"
```

Comment:
The source file has a `-go` suffix. Change to:
```
install -DT "./cmd/device-rfid-llrp-go" "$SNAPCRAFT_PART_INSTALL/bin/device-rfid-llrp"
```

Reply:
I have made the changes you mentioned

Reply:
I only suggested a single suffix addition for the executable name. Your additional changes break the snap. Please revert the additional `-go` additions.

Reply:
Thank you, I removed the additional -go

Reply:
There were many more changes on snap/snapcraft.yaml: https://github.com/edgexfoundry/device-rfid-llrp-go/pull/52/commits/663551aaef05a646e9206bf7a518f723bf6440fe

Let's get this PR merged and I'll make another PR to fix it. Thanks.